### PR TITLE
Use microsoft/dotnet:2.1-sdk in CI

### PR DESCRIPTION
**Pull request by zacharycmontoya**
Changes proposed in this pull request:

Current smoke test issues may be alleviated by using a non-Alpine distro. Move off the Alpine distribution to a Debian distribution of the .NET Core SDK docker image.

@DataDog/apm-dotnet

### CODE REVIEW

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs [hunk 1]
```diff
@@ -23,5 +24,12 @@ public void NoExceptions()
 
             CheckForSmoke();
         }
+
+        protected override void AssertProcessResultIsSuccessful(ProcessResult result)
+        {
+            var successCodes = new HashSet<int> { 0, 139 };
```
Comment (zacharycmontoya) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs]: Oftentimes the StackExchange smoke test exits with a 139 exit code. If we see that exit code and nothing being written to standard error, I think we should consider the result passing. Thoughts?

Comment (colin-higgins) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs]: I don't think I agree. My first instinct was the same, but after talking with @lucaspimentel about it, it's something I think we need to look into.

Comment (zacharycmontoya) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs]: I would ultimately like to get to the bottom of this too, but it is a little frustrating that it seems to reproduce very frequently and our managed code is invoked just fine. I looked up the segfault issue and I found a possible related CoreCLR issue concerning a race condition that particularly affects Alpine (we're using the 2.1 SDK Alpine image). I don't really want to investigate this further at the moment, but perhaps we can test that hypothesis when the next 2.1 patch comes out.

Comment (lucaspimentel) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs]: I also found Alpine mentioned when I was searching for 139 and corefx segfaults in docker. We can easily change the base docker image in `docker/dotnet.dockerfile`, if you want to try that. See https://hub.docker.com/_/microsoft-dotnet-core-sdk/

**Comment by zacharycmontoya** [comment]
@lucaspimentel @colin-higgins I changed the dotnet.docker file to use the `microsoft/dotnet:2.1-sdk` image as the base image, and the StackExchange smoke test didn't fail on two consecutive runs. What do you think about merging this change in?

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs (hunk 1 above)
Comment (lucaspimentel) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/SmokeTests/StackExchangeRedisStackOverflowExceptionSmokeTest.cs]: https://github.com/dotnet/coreclr/issues/22176

File: docker/dotnet.dockerfile [hunk 2]
```diff
@@ -1,6 +1,4 @@
-FROM microsoft/dotnet:2.1-sdk-alpine
-
-RUN apk add bash libc6-compat
+FROM microsoft/dotnet:2.1-sdk
```
Comment (lucaspimentel) [inline docker/dotnet.dockerfile]: I think the recommended way to do this nowadays is:
```suggestion
FROM mcr.microsoft.com/dotnet/core/sdk:2.1
```

Related: we should think about how to test on 2.1, 2.2, and 3.0 when it's released next month.

**Pull request merged**